### PR TITLE
[BUGFIX] Improve null handling to avoid NullPointerException and ext…

### PR DESCRIPTION
**Pull request by manuelsblanco**
…ract the method 'getAuthenticator'.

**Thanks for contributing to Selenium!**
**A PR well described will help maintainers to quickly review and merge it**

Before submitting your PR, please check our [contributing](https://github.com/SeleniumHQ/selenium/blob/trunk/CONTRIBUTING.md) guidelines.
Avoid large PRs, help reviewers by making them as simple and short as possible.

### Description
As requested by @titusfortner, refactor the class JdkHttpClient

### Motivation and Context
Some validations were always false, and I'm handling null possibilities to avoid NullPointerExceptions.

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Checklist

- [x] I have read the [contributing](https://github.com/SeleniumHQ/selenium/blob/trunk/CONTRIBUTING.md) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by titusfortner** [comment]
What I suggested was some of the null checks can be replaced with something more concise like @mykola-mokhnach
 pointed out in a previous PR

```
ofNullable(config.version()).map(Version::valueOf).ifPresent(builder::version)
```

File: java/src/org/openqa/selenium/remote/http/jdk/JdkHttpClient.java [hunk 1]
```diff
@@ -103,8 +103,13 @@ public class JdkHttpClient implements HttpClient {
           new Authenticator() {
             @Override
             protected PasswordAuthentication getPasswordAuthentication() {
-              return new PasswordAuthentication(username, password.toCharArray());
+              if (username != null && password != null) {
```
Comment (mykola-mokhnach) [inline java/src/org/openqa/selenium/remote/http/jdk/JdkHttpClient.java]: how about 

```
if (username == null || password == null) {
  return null;
}
return new PasswordAuthentication(username, password.toCharArray());
```

File: java/src/org/openqa/selenium/remote/http/jdk/JdkHttpClient.java [hunk 2]
```diff
@@ -129,10 +134,7 @@ protected PasswordAuthentication getPasswordAuthentication() {
           new ProxySelector() {
             @Override
             public List<Proxy> select(URI uri) {
-              if (proxy == null) {
-                return List.of();
-              }
-              if (uri.getScheme().toLowerCase().startsWith("http")) {
+                if (uri.getScheme().toLowerCase().startsWith("http")) {
```
Comment (mykola-mokhnach) [inline java/src/org/openqa/selenium/remote/http/jdk/JdkHttpClient.java:136]: broken indentation

Comment (titusfortner) [reply java/src/org/openqa/selenium/remote/http/jdk/JdkHttpClient.java:136]: Run `./scripts/formt.sh` to fix all formatting. RBE build will fail without that.

**Comment by manuelsblanco** [comment]
> What I suggested was some of the null checks can be replaced with something more concise like @mykola-mokhnach pointed out in a previous PR
> 
> ```
> ofNullable(config.version()).map(Version::valueOf).ifPresent(builder::version)
> ```

Code updated with the @mykola-mokhnach  suggestion's.

File: java/src/org/openqa/selenium/remote/http/jdk/JdkHttpClient.java (hunk 1 above)
Comment (diemol) [reply java/src/org/openqa/selenium/remote/http/jdk/JdkHttpClient.java]: I don't see this suggestion getting applied, @manuelsblanco.

**Review by diemol — CHANGES_REQUESTED** [review]

File: java/src/org/openqa/selenium/remote/http/jdk/JdkHttpClient.java (hunk 1 above)
Comment (manuelsblanco) [reply java/src/org/openqa/selenium/remote/http/jdk/JdkHttpClient.java]: @diemol Sorry I was a couple of days out, getting applied now.

Comment (manuelsblanco) [reply java/src/org/openqa/selenium/remote/http/jdk/JdkHttpClient.java]: @diemol The code was updated with the suggestions

**Comment by diemol** [comment]
@manuelsblanco I am sorry it has taken a while to merge this, can you please resolve the conflicts and I can merge this.

**Pull request closed without merging**